### PR TITLE
docs: update proxy-rewrite plugin example

### LINKED ISSUE #11004: docs: proxy-rewrite plugin http://iresty.com/test/home.html 404
### Current State

in [doc](https://apisix.apache.org/zh/docs/apisix/plugins/proxy-rewrite/) ,http://iresty.com/test/home.html 404

### Desired State

The url should be accessible

### PR DESCRIPTION
**Pull request by sunheyi6**
### Description

Fixes #11004

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Review by shreemaan-abhishek — APPROVED** [review]

File: docs/en/latest/plugins/proxy-rewrite.md [hunk 1]
```diff
@@ -57,67 +57,65 @@ Header configurations are executed according to the following priorities:
 The example below enables the `proxy-rewrite` Plugin on a specific Route:
 
 ```shell
-curl http://127.0.0.1:9180/apisix/admin/routes/1  -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
-{
+curl "http://127.0.0.1:9180/apisix/admin/routes" -X PUT \
+  -H "X-API-KEY: ${ADMIN_API_KEY}" \
```
Comment (kayx23) [inline docs/en/latest/plugins/proxy-rewrite.md:61]: Not sure if this should be copied from API7 docs

**Review by kayx23 — COMMENTED** [review]
If the purpose of this PR is to close this [issue](https://github.com/apache/apisix/issues/11004) (which has been manually closed), modifying the examples this way does not change anything.

**Comment by sunheyi6** [comment]
> If the purpose of this PR is to close this [issue](https://github.com/apache/apisix/issues/11004) (which has been manually closed), modifying the examples this way does not change anything.

The purpose is to repair the document and make it easier for new people to read the document.

File: docs/en/latest/plugins/proxy-rewrite.md (hunk 1 above)
Comment (sunheyi6) [reply docs/en/latest/plugins/proxy-rewrite.md:61]: ADMIN_API_KEY is actually edd1c9f034335f136f87ad84b625c8f1, so I think there is no problem

Comment (kayx23) [reply docs/en/latest/plugins/proxy-rewrite.md:61]: I wrote the API7 docs and it's meant to be a user defined env var 😅

**Comment by kayx23** [comment]
So could you clarify what's "repaired"?

**Comment by sunheyi6** [comment]
> * If you actually run the example, you won't get the expected response. It doesn't address the issue you opened earlier.

Yes, I actually ran it and it was indeed not the expected result.The actual results are as follows

```shell
shy@LAPTOP-OJLO0M82:~$ curl "http://127.0.0.1:9080/headers"
{
  "headers": {
    "Accept": "*/*",
    "Host": "httpbin.org",
    "User-Agent": "curl/7.68.0",
    "X-Amzn-Trace-Id": "Root=1-65efbcd4-2398c1e53ce9ff957b63c1fb",
    "X-Forwarded-Host": "127.0.0.1"
  }
}
shy@LAPTOP-OJLO0M82:~$ curl "http://127.0.0.1:9080/headers"
{
  "headers": {
    "accept": "*/*",
    "accept-encoding": "gzip, br",
    "cf-connecting-ip": "45.194.20.22",
    "cf-ipcountry": "HK",
    "cf-ray": "863053e13ff8e68a",
    "cf-visitor": "{\"scheme\":\"https\"}",
    "connection": "Keep-Alive",
    "content-type": "application/json",
    "host": "mock.api7.ai",
    "user-agent": "curl/7.68.0",
    "x-application-owner": "API7.ai",
    "x-forwarded-for": "172.19.0.1",
    "x-forwarded-host": "127.0.0.1",
    "x-forwarded-port": "9080",
    "x-forwarded-proto": "https",
    "x-real-ip": "45.194.20.22",
    "X-Application-Owner": "API7.ai",
    "Content-Type": "application/json"
  }
```
I'm not sure why it's happening. One of the reasons I raised this PR is to ask for help understanding why it's like this. Of course, I didn't explain it clearly at the beginning, which is my fault. Sorry.

**Comment by kayx23** [comment]
How come you get different results form sending the same request twice? Configured two upstream nodes?

**Comment by sunheyi6** [comment]
> How come you get different results form sending the same request twice? Configured two upstream nodes?

```shell
shy@LAPTOP-OJLO0M82:~$ curl -X GET http://localhost:9180/apisix/admin/upstreams
{"total":0,"list":[]}
```

**Comment by kayx23** [comment]
The only explanation is you have configured two upstream nodes in your route, and the requests were forwarded to different nodes.

As for this PR, it doesn't improve the example in a material way, so I'm closing this for now.

**Comment by sunheyi6** [comment]
> * If you actually run the example, you won't get the expected response. It doesn't address the issue you opened earlier.

@kayx23 I have switched to a new Windows computer and the tests are giving the expected results in wsl2.
The order is as follows: test  add test delete add test
```shell
root@DESKTOP-F0LS87N:~# curl "http://127.0.0.1:9080" --head | grep Server
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
Server: APISIX/3.8.0
root@DESKTOP-F0LS87N:~# curl -X GET http://localhost:9180/apisix/admin/upstreams
{"total":0,"list":[]}
root@DESKTOP-F0LS87N:~# curl "http://127.0.0.1:9080/headers"
{"error_msg":"404 Route Not Found"}
root@DESKTOP-F0LS87N:~# curl "http://127.0.0.1:9180/apisix/admin/routes" -X PUT \
X-API-KEY: ${ADM>   -H "X-API-KEY: ${ADMIN_API_KEY}" \
>   -d '{
 "proxy-rewrite->     "id": "proxy-rewrite-route",
>     "methods": ["GET"],
>     "uri": "/headers",
>     "plugins": {
>       "proxy-rewrite": {
>         "host": "myapisix.demo"
>       }
>     },
>     "upstream": {
>       "type": "roundrobin",
>       "nodes": {
>         "httpbin.org:80": 1
>       }
>     }
>   }';
{"key":"/apisix/routes/proxy-rewrite-route","value":{"status":1,"create_time":1710499147,"upstream":{"pass_host":"pass","nodes":{"httpbin.org:80":1},"hash_on":"vars","type":"roundrobin","scheme":"http"},"id":"proxy-rewrite-route","priority":0,"plugins":{"proxy-rewrite":{"use_real_request_uri_unsafe":false,"host":"myapisix.demo"}},"update_time":1710499147,"uri":"/headers","methods":["GET"]}}
root@DESKTOP-F0LS87N:~# curl "http://127.0.0.1:9080/headers"
{
  "headers": {
    "Accept": "*/*",
    "Host": "myapisix.demo",
    "User-Agent": "curl/7.68.0",
    "X-Amzn-Trace-Id": "Root=1-65f42550-22507dd94115831074d1f25d",
    "X-Forwarded-Host": "127.0.0.1"
  }
}
root@DESKTOP-F0LS87N:~# curl "http://127.0.0.1:9180/apisix/admin/routes" -X PUT \
X-API-KEY: ${ADM>   -H "X-API-KEY: ${ADMIN_API_KEY}" \
>   -d '{
>     "id": "proxy-rewrite-route",
>     "methods": ["GET"],
>     "uri": "/headers",
>     "plugins": {},
>     "upstream": {
>       "type": "roundrobin",
>       "nodes": {
 "127.0.0.1:80":>         "127.0.0.1:80": 1
>       }
>     }
>   }';
{"key":"/apisix/routes/proxy-rewrite-route","value":{"status":1,"create_time":1710499147,"upstream":{"pass_host":"pass","nodes":{"127.0.0.1:80":1},"hash_on":"vars","type":"roundrobin","scheme":"http"},"id":"proxy-rewrite-route","priority":0,"plugins":{},"update_time":1710499244,"uri":"/headers","methods":["GET"]}}
root@DESKTOP-F0LS87N:~# curl "http://127.0.0.1:9080/headers"
<html>
<head><title>502 Bad Gateway</title></head>
<body>
<center><h1>502 Bad Gateway</h1></center>
<hr><center>openresty</center>
<p><em>Powered by <a href="https://apisix.apache.org/">APISIX</a>.</em></p></body>
</html>
root@DESKTOP-F0LS87N:~# curl "http://127.0.0.1:9180/apisix/admin/routes" -X PUT   -H "X-API-KEY: ${ADMIN_API_KEY}"   -d '{
    "id": "proxy-rewrite-route",
    "methods": ["GET"],
    "uri": "/headers",
    "plugins": {
      "proxy-rewrite": {
        "host": "myapisix.demo"
      }
    },
    "upstream": {
      "type": "roundrobin",
      "nodes": {
        "httpbin.org:80": 1
      }
    }
  }';
{"key":"/apisix/routes/proxy-rewrite-route","value":{"status":1,"create_time":1710499147,"upstream":{"pass_host":"pass","nodes":{"httpbin.org:80":1},"hash_on":"vars","type":"roundrobin","scheme":"http"},"id":"proxy-rewrite-route","priority":0,"plugins":{"proxy-rewrite":{"use_real_request_uri_unsafe":false,"host":"myapisix.demo"}},"update_time":1710499254,"uri":"/headers","methods":["GET"]}}
root@DESKTOP-F0LS87N:~# curl "http://127.0.0.1:9080/headers"
{
  "headers": {
    "Accept": "*/*",
    "Host": "myapisix.demo",
    "User-Agent": "curl/7.68.0",
    "X-Amzn-Trace-Id": "Root=1-65f425b9-6aea290363a3d9e724620d9e",
    "X-Forwarded-Host": "127.0.0.1"
  }
}
root@DESKTOP-F0LS87N:~#
```

**Pull request closed without merging**